lib/utils.pm

```diff
@@ -1094,7 +1094,7 @@ sub set_hostname {
     systemctl 'status network.service';
     save_screenshot;
 
-    if (systemctl('is-active NetworkManager', ignore_failure => 1) == 0) {
+    if (get_var('USE_SUPPORT_SERVER') && systemctl('is-active NetworkManager', ignore_failure => 1) == 0) {
```

Comment:
this is a lib module and i am concerned about the fact that such a function has a depended condition to `USE_SUPPORT_SERVER`. there is no quarantine that multimachine jobs will use a supportserver. 
What test case does this consider?
is it possible to avoid this and make the function job-variable-free?